src/sparseml/optim/modifier.py

```diff
@@ -294,8 +294,11 @@ def load_framework_list(yaml_str: str, framework: str):
             modifiers = [container]
         elif isinstance(container, List):
             modifiers = container
-        else:
-            modifiers = container["modifiers"]
+        else:  # Dict
+            modifiers = []
+            for key, modifier_list in container.items():
+                if "modifiers" in key and isinstance(modifier_list, List):
```

Comment:
if a user accidentally mispells a group, we'll have a silent ignore here. We should at least warn if a key doesn't have modifiers in it and won't be used, and potentially could throw an exception here to be explicit since we aren't supporting anything else currently

Reply:
agreed. this was definitely a concern the issue is that not every top level dict element will necessarily be a modifier list.  We could have for instance a list of parameter names stored as a variable to reference elsewhere.  If this wasn't the case I would not have required "modifier" to be in the name at all.

One alternative is we could search all lists for `BaseModifier` objects and flatten that way, letting group names just be for organization. Will message in Slack.

Reply:
Actually, will raise an error when a `Modifier` object is found outside of a named "modifier" list.  That way we could keep the YAML flexibility, but enforce some simple structure in how modifiers are defined.